### DISCUSSION TITLE
Edamame row output question

### DISCUSSION THREAD

**Discussion by robert-stuttaford** [borkdude/edamame#80] [Q&A]
Using edamame to parse EDN and report on errors (which is rad), noticed a discrepancy. 

(I recognise this may be totally correct behaviour).

```clojure
 1 {:metadata
 2 {:title "Turn Strategy to Action"} 
 3  
 4 :default-set "1"
 5 :chat-avatar "avatar11-v3.jpg" 
 6 :icon "cog-icon.png"
 7 :no-icon-border? true
 8 
 9 :variables 
10 (                      
11 { ;; <==== here's the issue i'm working to provide the user with feedback on
12 {:type :user
13  :name "did_you_write_anything" 
14  :label "Did the learner write anything?"
15  :export-to-profile? true
16  :default-value 0}   
17   
18 )
19 }
```

Here's the output - correct `:row` of 11 is inside the `:edamame.impl.parser/opened-delimiter` key, but the row of the end of the containing seq (18) is in the top level `:row`.

I'm hoping to simply use the top level `:row` value in guiding the user, and, ideally, in this case, it'd be great if it was that 11 from the inner map.

```clojure
;; (:ex-data err)
{ :end-location true
  :end-row-key :end-row
  :row-key   :row
  :col-key   :col
  :source-key :source
  :edamame.impl.parser/expected-delimiter "}"
  :type      :edamame/error
  :edamame.impl.parser/opened-delimiter { :char      "{"
                                          :row       11
                                          :col       2 }
  :col       1
  :end-col-key :end-col
  :row       18 }
```

**Comment by borkdude** [comment]
@robert-stuttaford I don't understand the question. Can you give a minimal repro in working code, with actual + expected output?

**Reply by robert-stuttaford** [reply]
Happy to do that, but here's an alternative phrasing which might be simpler to tackle:

Is it safe to depend on `:edamame.impl.parser/opened-delimiter` key as part of its API?

**Reply by borkdude** [reply]
Ah I see. The `.impl` part in the name suggests no, but we can make it public.

**Reply by borkdude** [reply]
I see we already did such a thing somewhere in the code: `:edamame/opened-delimiter (str opened)`. 

So can you give me a repro where this delimiter isn't public while it should be? Might be an oversight.

**Comment by robert-stuttaford** [comment]
Simple repro:

```clojure
{
 :x (
     { ;; offending error
     {:a 1}
     )
 }
```

`(:ex-data e)` with the nil values removed:

```clojure
{ :end-location true
  :end-row-key :end-row
  :row-key   :row
  :col-key   :col
  :source-key :source
  :edamame.impl.parser/expected-delimiter "}"
  :type      :edamame/error
  :edamame.impl.parser/opened-delimiter { :char      "{"
                                          :row       3
                                          :col       6 }
  :col       6
  :end-col-key :end-col
  :row       6 }
```

**Reply by robert-stuttaford** [reply]
Does this repro suffice or do you need something more, @borkdude ? 😃

**Reply by borkdude** [reply]
Opened an issue for it here: https://github.com/borkdude/edamame/issues/81

**Reply by borkdude** [reply]
@robert-stuttaford Do you need to access the location of the opened delimiter or is the opened delimiter character alone sufficient?

**Reply by robert-stuttaford** [reply]
I need the location data; I'm using it to issue a selection to a codemirror instance, so that when users click the error, the offending text is selected

**Reply by borkdude** [reply]
Can you try if this commit solved your problem? https://github.com/borkdude/edamame/commit/17479c6e93808534f4123ca93944c7f637a6ebf2

**Reply by robert-stuttaford** [reply]
This is perrrrrrrrfect, thank you for jumping on this so quickly!

**Reply by borkdude** [reply]
OK, merged to master. I assume you are using deps.edn, else let me know if you need a clojars release.
